tasks/java/jvm/permissions_nix.go

```diff
@@ -76,674 +87,379 @@ func (p JavaJVMPermissions) Explain() string {
+		}
+		for _, process := range sysPropsProcesses {
+			if process.ProcID == proc.Pid {
+				nrTempDirVal, isTempDirPresent := process.SysPropsKeyToVal[nrTempDirKey]
+				if isTempDirPresent {
+					tempDir = nrTempDirVal
+					tempDirSource = nrTempDirKey
 				} else {
-					log.Debug(taskName + "Unable to determine the cwd or directory containing nrdiag executable. If you are using the system property newrelic.logfile to set the newrelic log file name, you should manually check that the log file is created properly.")
+					javaTmpDirVal, isTmpDirPresent := process.SysPropsKeyToVal[javaTmpDirKey]
+					if isTmpDirPresent {
+						tempDir = javaTmpDirVal
+						tempDirSource = javaTmpDirKey
+					}
 				}
 			}
 		}
-		if newValues.TempDir.Value == "" {
-			/* -Dnewrelic.TempDir=/path/to/temp/dir */
-			newValues.TempDir = valueSource{checkForSingleProp("-Dnewrelic.tempdir=(.+)", arg), sysPropSource}
-			log.Debug(taskName + "Setting temp dir location to " + newValues.TempDir.Value + " from " + newValues.TempDir.Source)
-		}
-		/* -Djava.io.tmpdir=/Path/to/temp */
-		/* ensure this location hasn't already been set with -Dnewrelic.TempDir, which has greater precedence */
-		if newValues.TempDir.Value == "" {
-			newValues.TempDir = valueSource{checkForSingleProp("-Djava.io.tmpdir=(.+)", arg), sysPropSource}
-			log.Debug(taskName + "Setting temp file location to " + newValues.TempDir.Value + " from " + newValues.TempDir.Source)
-		}
-	}
-	return newValues
-}
-
-func checkForSingleProp(regex string, key string) (value string) {
-	reg := regexp.MustCompile(regex)
-	match := reg.FindStringSubmatch(key)
-	if len(match) != 0 {
-		return match[1]
 	}
-	return value
-}
-
-func checkForYAMLValues(values agentValues) (newYAMLFile, newLogFileName, newLogPath valueSource, err error) {
-	newYAMLFile = values.YamlFile
-	newLogFileName = values.LogFileName
-	newLogPath = values.LogPath
-
-	/* if YAML file location isn't set by sys prop, it should be in default location */
-	if newYAMLFile.Value == "" {
-		/* default YAML file location is in dir containing Agent JAR */
-		newYAMLFile = valueSource{(filepath.Dir(values.AgentJarLocation.Value) + "/newrelic.yml"), yamlSource}
-		log.Debug(taskName + "Setting yaml file location to " + newYAMLFile.Value + " from " + newYAMLFile.Source)
+	//if none of those system properties are set, check the operating system's default tmp dir
+	if len(tempDir) == 0 {
+		tempDir = os.TempDir() //may return a directory that does not exist but soon we'll find out as we check for this directory's permissions
+		tempDirSource = "the default tmp directory for the Operation System"
 	}
-	/* YAML file variable checks */
-	if newLogFileName.Value == "" {
+	//start assigning javaAgentPermissions values to temp directory
+	j.TempFilesCanCreate.Source = tempDirSource
+	j.TempFilesCanCreate.Value = tempDir
 
-		rawFileName, err := tasks.ReturnLastStringSubmatchInFile("log_file_name: (.+)", newYAMLFile.Value)
```

Comment:
ReturnLastStringSubmatchInFile searches a file for any string match, regardless if it is a comment or not. It's removal solves for finding a remarked line in the yaml 🥳

Reply:
You are so right! https://github.com/newrelic/newrelic-diagnostics-cli/blob/main/tasks/fileSearch.go#L92
I wonder if the author of the task was aware of this effect.  I am guessing "no" because this is not how I would judge a current path to an existing log file. If a user comments out a key/value in a new relic config file, I would assume that the user is no longer storing the log file in that path.

Reply:
Looks like this utility serves other functions, just used once off to grab a yaml key.